### ISSUE TITLE
what I would wish for

### ISSUE THREAD

**Issue by Krutok** [trickerer/Trinity-Bots#194]
I would wish if the gear could be saved in the database, because if you have to keep the gear for the bots depending on the content, then at some point it will fill up the bank or inventory.

Therefore, once the bot has owned an item, it stores the item in the database and you can then equip it whenever you get a bot again.

So that you can farm gear sets for the bots, so to speak, and when you talk to the bot, you select the appropriate items in the menu and equip them to the bot.

Should be possible I think.
Oh and the enrage, if you dismiss the bot I don't think it's that great, it would be nice if you could turn it off in the config if you don't want it.

**Comment by Krutok** [comment]
Thanks for the implementation with the Enrage Option :-)
Now it would be great if you could save the gear that you give the bots in the database so that you don't have to equip a whole raid again and again, but once the bot has the item on it becomes an item in for the bot saved to the database.

**Comment by Krutok** [comment]
Is it possible to save the equipment in the database or rather not? would be really nice if that were possible, then you no longer have all the equipment in the bank and in the inventory

**Comment by trickerer** [comment]
It's possible, that's not a small task though. There are some related problems not resolved also.
No matter how big bot bags are, at dismiss bot will still have to give all their gear back. There is a pretty serious sneaky bug where items sent through mail by bots sometimes don't have their owner set properly, this makes players unable to retrieve some items from mail, the only way is to destroy those inaccessible items.

**Comment by Riveir** [comment]
Based on the title of this post I thought I would throw in my two cents. Not sure if it would be possible at some point to add a feature to rename your npcbot. If not it's obviously not a deal breaker.

**Comment by trickerer** [comment]
Well I performed a couple quicks tests and it seems like you indeed can force feed client creature info, including custom name. The problem is the delivery system, I don't yet see a way to make client actually ask for new info, which would solve this problem almost entirely.

**Comment by Riveir** [comment]
> Well I performed a couple quicks tests and it seems like you indeed can force feed client creature info, including custom name. The problem is the delivery system, I don't yet see a way to make client actually ask for new info, which would solve this problem almost entirely.

Oh I guess I was thinking that because there was already a system to rename pets that there might actually be a system that could be utilized. It's not a huge thing. Can you tell me is there just a couple entries in the db that need to be changed to manually do this? Or would that even work?

**Comment by trickerer** [comment]
No, you can't just rename bot in the DB and call it good. Unlike pets (maybe), normal creatures' names are always cached by the client.
I don't think client sends pet name requests for npcbots. I'm not even sure if renamed pet's name is properly updated for other players. More research is needed.

**Comment by Riveir** [comment]
> No, you can't just rename bot in the DB and call it good. Unlike pets (maybe), normal creatures' names are always cached by the client. I don't think client sends pet name requests for npcbots. I'm not even sure if renamed pet's name is properly updated for other players. More research is needed.

![Wow 2022-10-09 20-06-13](https://user-images.githubusercontent.com/96812524/194794286-dfe82df3-f110-462d-a275-2cca779eaf22.png)
I'm not sure if this is going to cause issues down the road. I'm going to play with it a bit but I was able to do the db edits for both name and model id the contingent is that I needed to clear my client cache to see the change in game.

**Comment by Codewow** [comment]
I’ll second the OP. This mod is fantastic and I really appreciate your work on this. Saving gear to a bot per player would be a great addition to this since I just found out the hard way that they all toss their gear back at me when I dismissed them while messing around. 

Thank you again.

**Comment by Codewow** [comment]
> > I’ll second the OP. This mod is fantastic and I really appreciate your work on this. Saving gear to a bot per player would be a great addition to this since I just found out the hard way that they all toss their gear back at me when I dismissed them while messing around.
> > Admittedly I only skimmed the docs. I see it in there about this. Hopefully, we can get this as a feature though. Would be great for setting up raid nights with my 5 friends on my server lol.
> > Thank you again.
> 
> Hey, it's not obvious in docs but ".npcb remove" dismisses the bot and keeps gear equipped on him.

Dismisses and allows someone else to use them or just makes them go back to their spawn and not follow you temporarily?

**Comment by trickerer** [comment]
> > Hey, it's not obvious in docs but ".npcb remove" dismisses the bot and keeps gear equipped on him.

> Dismisses and allows someone else to use them or just makes them go back to their spawn and not follow you temporarily?

Dismisses completely. This is a GM command. Temporary dismiss is `.npcbot command unbind`

**Comment by westernoutlaw** [comment]
Would it be possible to add mage bots ability to cast portals on request, respective to their level?

**Comment by trickerer** [comment]
Yes, but every time I think about this I remember how many of teleporter mods exist, even pure SQL ones.

**Comment by Krutok** [comment]
I hope there will be a feature to save bot gear. Because if you take the bots on raids or on battlefields, you can't put as much gear on the bench
